cl/citations/api_views.py

```diff
@@ -0,0 +1,209 @@
+from http import HTTPStatus
+
+import eyecite
+from asgiref.sync import async_to_sync
+from django.db.models import QuerySet
+from django.http import HttpResponse
+from django.template.defaultfilters import slugify
+from reporters_db import EDITIONS
+from rest_framework.exceptions import NotFound
+from rest_framework.generics import GenericAPIView
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from cl.api.pagination import MediumAdjustablePagination
+from cl.citations.api_serializers import CitationRequestSerializer
+from cl.citations.exceptions import MultipleChoices
+from cl.citations.utils import get_canonicals_from_reporter
+from cl.search.api_serializers import OpinionClusterSerializer
+from cl.search.models import OpinionCluster
+from cl.search.selectors import get_clusters_from_citation_str
+
+SLUGIFIED_EDITIONS: dict[str, str] = {
+    str(slugify(item)): item for item in EDITIONS.keys()
+}
```

Comment:
If you're going to put this here, you should replace it at `opinion_pages.views:1146` too. DRY.